app/src/main/kotlin/com/wire/android/mapper/RegularMessageContentMapper.kt

```diff
@@ -234,3 +269,19 @@ private fun String?.orUnknownName(): UIText = when {
     this != null -> UIText.DynamicString(this)
     else -> UIText.StringResource(R.string.username_unavailable_label)
 }
+
+private fun mapRecipientsFailure(userList: List<User>, deliveryStatus: DeliveryStatus?): DeliveryStatusContent {
+    return when (deliveryStatus) {
+        is DeliveryStatus.PartialDelivery -> DeliveryStatusContent.PartialDelivery(
+            failedRecipients = deliveryStatus.recipientsFailedDelivery
+                .map { userId -> UIText.DynamicString(userList.findUser(userId = userId)?.name.orEmpty()) }
```

Comment:
We could use:
```
userList.findUser(userId = userId)?.name.orUnknownName()
```
This way we get `UIText.StringResource` when we put the generic name instead of real one and we could filter out it later more easily, I know it's an extreme case but what if someone chose to name his account "Name not available" - it would be filtered out 😅 Instead, when filtering this way:
```
filter { it is UIText.StringResource && it.resId == R.string.username_unavailable_label }
```
it would be more safe 😄